app/src/main/java/mysh/dev/gemcap/network/GeminiClient.kt

```
@@ -394,24 +436,58 @@ class GeminiClient(
+        val header = parseHeader(inputStream)
+        if (header.status in 20..29) {
+            val tempFile = File(outputFile.parent, outputFile.name + ".tmp")
+            try {
+                readResponseBodyToFile(inputStream, tempFile, onProgress)
+                if (!tempFile.renameTo(outputFile)) {
+                    tempFile.inputStream().use { input ->
+                        outputFile.outputStream().use { output -> input.copyTo(output) }
+                    }
+                    tempFile.delete()
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)